### ISSUE TITLE
Call `detach` when an error is thrown during async iteration of `iter()`

### ISSUE THREAD

**Issue by remorses** [garronej/evt#67]
Currently the `iter()` method is not calling `self.detach()` in case an error is thrown during async iteration. 

I am not an async iterator expert but i think there should be a `throw()` method to detach to prevent memory leaks

```ts
throw() {
  self.detach(ctx)
  throw error;
}
```

https://github.com/garronej/evt/blob/eddfe891308d8aac9c48c7c7ac9a5ec687ffa48b/src/lib/Evt.ts#L872

**Comment by garronej** [comment]
Hello @remorses,  

Thank you for pointing this out.  
I think you are right.

**Comment by garronej** [comment]
Hello @remorses,  

Sorry for the delay in dealing with this issue.  
I've merged your PR, I'm releasing a new version.  

Thank you for your contribution!

**Comment by remorses** [comment]
Thank you!
